src/ophyd_async/core/_signal.py

```diff
@@ -588,16 +588,15 @@ async def _wait_for_value():
                     break
```

Comment:
Just noticed that somewhere along the line we lost the ability to pass a callable as `match_value`, please could you add it back in?